src/lib.rs

```diff
@@ -403,6 +455,22 @@ pub fn setup_environment_variables(
     env_vars.push(("IDF_TOOLS_PATH".to_string(), instal_dir_string));
     let idf_path_string = idf_path.to_str().unwrap().to_string();
     env_vars.push(("IDF_PATH".to_string(), idf_path_string));
+    env_vars.push((
```

Comment:
While this implementation works now, in the future tools.json might include additional env variables that might need to be added.

I suggest an approach to obtain variables using `export_vars` field of the IDF Tool. For example in current ESP-IDF master on Windows there is an override to always install **CCache**, which contributes `IDF_CCACHE_ENABLE=1` and perform string replacements for string variables like `${TOOL_PATH}`

Reply:
that will be part of processing the file in order to offer optional tool down the roadmap.

Reply:
But CCache is not optional on Windows. There is a platform override section.